subprojects/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/GradleEnterprisePluginCheckInIntegrationTest.groovy

```diff
@@ -121,8 +124,36 @@ class GradleEnterprisePluginCheckInIntegrationTest extends AbstractIntegrationSp
         getMinimumPluginVersionForConfigurationCaching() | true
     }
 
+    @Requires(IntegTestPreconditions.NotConfigCached)
+    def "shows warning message when Gradle Enterprise plugin version is used with isolated projects enabled"() {
+        given:
+        plugin.runtimeVersion = plugin.artifactVersion = pluginVersion
+        applyPlugin()
+        settingsFile << """
+            println "present: " + services.get($GradleEnterprisePluginManager.name).present
+        """
+
+        when:
+        succeeds("t", "-D${StartParameterBuildOptions.IsolatedProjectsOption.PROPERTY_NAME}=true")
+
+        then:
+        output.contains("present: ${applied}")
+
+        and:
+        output.contains("gradleEnterprisePlugin.checkIn.unsupported.reasonMessage = $UNSUPPORTED_PLUGIN_DUE_TO_ISOLATED_PROJECTS_MESSAGE") != applied
+
+        where:
+        pluginVersion                                | applied
+        '3.11.4'                                     | false
+        getMinimumPluginVersionForIsolatedProjects() | true
```

Comment:
💅This could be tidier with Groovy's getters: `minimumPluginVersionForIsolatedProjects`